src/main/java/io/vertx/micrometer/backends/PrometheusBackendRegistry.java

```
@@ -0,0 +1,74 @@
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.vertx.micrometer.backends;
+
+import io.micrometer.core.instrument.MeterRegistry;
+import io.micrometer.prometheus.PrometheusConfig;
+import io.micrometer.prometheus.PrometheusMeterRegistry;
+import io.vertx.core.Vertx;
+import io.vertx.core.eventbus.EventBus;
+import io.vertx.core.http.HttpServer;
+import io.vertx.core.http.HttpServerOptions;
+import io.vertx.ext.web.Router;
+import io.vertx.micrometer.VertxPrometheusOptions;
+
+/**
+ * @author Joel Takvorian
+ */
+public final class PrometheusBackendRegistry implements BackendRegistry {
+  private final PrometheusMeterRegistry registry;
+  private final Vertx vertx;
+  private final VertxPrometheusOptions options;
+  private HttpServer server;
+
+  public PrometheusBackendRegistry(Vertx vertx, VertxPrometheusOptions options) {
+    this.vertx = vertx;
+    this.options = options;
+    registry = new PrometheusMeterRegistry(PrometheusConfig.DEFAULT);
+  }
+
+  @Override
+  public MeterRegistry getMeterRegistry() {
+    return registry;
+  }
+
+  @Override
+  public void eventBusInitialized(EventBus bus) {
+    if (options.isStartEmbeddedServer()) {
+      // Start dedicated server
+      HttpServerOptions serverOptions = options.getEmbeddedServerOptions();
+      if (serverOptions == null) {
+        serverOptions = new HttpServerOptions();
+      }
+      Router router = Router.router(vertx);
+      router.route(options.getEmbeddedServerEndpoint()).handler(routingContext -> {
+        String response = registry.scrape();
+        routingContext.response().end(response);
+      });
+      server = vertx.createHttpServer(serverOptions)
```

@tsegismont , looking for an advice here: I don't set any exception handler here (this is where we start the prometheus server), does it mean any exception will be silently discarded (which mean we should rather log it) or is it logged by default?

---

@tsegismont
I could reproduce some issues when setting `@Repeat(50)`, so I've put in place a retry mechanism and it seems ok now.

---

You can only get an exception when you call `listen`. Or do I miss something?